CMakeLists.txt

```diff
@@ -0,0 +1,169 @@
+  #add_subdirectory(profiling/roctx-connector)
+endif()
+
+if(DEFINED ENV{VTUNE_HOME})
+  set(VTune_ROOT $ENV{VTUNE_HOME})
+endif()
+if(VTune_ROOT)
+  #add_subdirectory(profiling/vtune-connector)
+  #add_subdirectory(profiling/vtune-focused-connector)
+else()
+  message(WARNING "Set VTUNE_HOME in environment or VTune_ROOT in build options to build VTune connectors")
+  set(VTune_ROOT "" CACHE STRING "Path to VTune Intel compiler")
+endif()
+
+# Note: Caliper docs https://software.llnl.gov/Caliper/build.html
+#       might miss some options, check the top CMakeLists.txt:
+#       https://github.com/LLNL/Caliper/blob/master/CMakeLists.txt
+macro(configure_caliper)
+  #set(BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS}) # Build shared (ON - default) or static library (OFF).
+  #set(BUILD_DOCS OFF)   # Build documentation.
+  #set(BUILD_TESTING OFF) # Build unit tests.
+  #set(WITH_FORTRAN OFF) # Build and install Fortran wrappers.
+  set(WITH_KOKKOS ON) # Enable Kokkos profiling support
+
+  # Build with support for the OpenMP tools interface.
+  if(OpenMP_FOUND)
+    set(WITH_OMPT ON)
+  else()
+    set(WITH_OMPT OFF)
+  endif()
+
+  set(WITH_SAMPLER ON)  # Enable time-based sampling on Linux.
+  set(WITH_TOOLS ON)    # Build Caliper’s tools (i.e, cali-query and mpi-caliquery). Default: On.
+  set(WITH_MPI ${KOKKOS_ENABLE_MPI}) # Build with MPI support.
+  set(WITH_ROCM ${Kokkos_ENABLE_HIP}) # Enable AMD ROCtracer/RocTX support
+  set(WITH_NVTX ${Kokkos_ENABLE_CUDA}) # Build adapters to forward Caliper annotations to NVidia’s nvtx annotation API. Set CUDA_TOOLKIT_ROOT_DIR to the CUDA installation.
+  set(WITH_CUPTI ${Kokkos_ENABLE_CUDA}) # Enable support for CUDA performance analysis
+                                        # (wrapping of driver/runtime API calls and CUDA activity tracing).
+  # Enable PAPI hardware counter service (requires papi)
+  if(PAPI_FOUND)
+    set(WITH_PAPI ON)
+    set(PAPI_PREFIX ${PAPI_ROOT})
+  endif()
+  # Build adapters to forward Caliper annotations to Intel’s VTune annotation API.
+  if(VTune_ROOT)
+    set(WITH_VTUNE ON)
+    set(ITT_PREFIX ${VTune_ROOT}/sdk)
+    # TODO: Caliper doesn't seek ITT in ${VTune_ROOT}/sdk/lib64 or ${VTune_ROOT}/sdk/lib32 (only in .../lib)
+  else()
+    set(WITH_VTUNE OFF)
+  endif()
+endmacro()
+
+# Note: Apex docs http://uo-oaciss.github.io/apex/install/#standalone_installation
+macro(configure_apex)
+  #set(BUILD_STATIC_EXECUTABLES OFF)
+  set(APEX_WITH_PAPI ${PAPI_FOUND})
+  set(APEX_WITH_CUDA ${Kokkos_ENABLE_CUDA})
+  set(APEX_WITH_BFD ON)
+  #set(BFD_ROOT ...)
+  if(OMPT_FOUND)
+    set(APEX_WITH_OMPT ON)
+    # TODO: set(OMPT_ROOT ...) ?
+  else()
+    set(APEX_WITH_OMPT OFF)
+  endif()
+  # TODO: Apex tries to fetch TPLs and install them in /usr/lib
+  #       (normally failing) unless we disable them...
+  set(APEX_WITH_ACTIVEHARMONY OFF)
+  set(APEX_WITH_OTF2 OFF)
+  set(APEX_WITH_MPI OFF)
+endmacro()
+
+# Build TPLs
+if(KOKKOSTOOLS_ENABLE_CALIPER)
+  configure_caliper()
```

Comment:
I don't really see the need for a macro here. Personally, i think you should just put setting the initial cache values in this if block.

Reply:
My motivation behind those macros was to move Caliper and Apex setup to the other file and keep scripts tight (applied in 4b7a3441904710e7c6e50952bc265eca83380b97) - let me know if you don't like that approach.